pkg/cloud/azure/actuators/machine/actuator_test.go

```diff
@@ -470,6 +472,44 @@ func TestAvailabilityZones(t *testing.T) {
 	}
 }
 
+func TestCustomUserData(t *testing.T) {
+	fakeScope := newFakeScope(t, v1alpha1.Node)
+	userDataSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testCustomUserData",
+			Namespace: "dummyNamespace",
+		},
+		Data: map[string][]byte{
+			"userData": []byte("test-userdata"),
+		},
+	}
+	fakeScope.CoreClient = controllerfake.NewFakeClient(userDataSecret)
+	fakeScope.MachineConfig.UserDataSecret = &corev1.SecretReference{Name: "testCustomUserData"}
+	fakeReconciler := newFakeReconcilerWithScope(t, fakeScope)
+	fakeReconciler.virtualMachinesSvc = &FakeVMCheckZonesService{}
+	if err := fakeReconciler.Create(context.Background()); err != nil {
+		t.Errorf("expected create to succeed %v", err)
+	}
+
+	userData, err := fakeReconciler.getCustomUserData()
+	if err != nil {
+		t.Errorf("expected get custom data to succeed %v", err)
+	}
+
+	if userData != "test-userdata" {
```

Comment:
This only tests the positive - can we create (perhaps table-driven tests) for success and expected failure?

Reply:
The tests do have negative tests right below this one. I did try to split out positive and negative, but didnt feel the need to 2 additional statements, i didnt cover one additional case of missing "userdata" field in secret, let me do that right away